### DISCUSSION TITLE
Table Column with url (Attachement table)

### DISCUSSION THREAD

**Discussion by Youssef-Elouasbi** [filamentphp/filament#11778] [Help]
### Package

Table builder

### Package Version

v3.1

### How can we help you?

Hi i have a table named "attachements" with a column called "file"  this column contain a link or url for this specific attachment like a pdf or image....
i need to show this field in my table inside the resource 
something like this :

TextColumn::make('nom')->label('Nom')->searchable()->sortable(),
     
but when the user click on that field the url should open 
help me please !

**Comment by leandrocfe** [comment]
You can use `->url()` in TextColumn

**Reply by jazzy1986** [reply]
work for one attachment, what if i got multiple attachments?

**Reply by krekas** [reply]
show as a list then?

**Reply by jazzy1986** [reply]
i think ->url() only accept string.

**Reply by jazzy1986** [reply]
->formatStateUsing work, but more dirty codes
